### DISCUSSION TITLE
Expose address book for external management

### DISCUSSION THREAD

**Discussion by hseg** [d99kris/nchat#189] [Ideas]
Currently, the contact sync uses the self-assigned contacts nicknames. While this works, sometimes the name given is unclear/useless for search (eg a relative running their business over their personal phone number appears under their business's name). Exposing/reading the address book from a separate file would allow managing these using eg `abook`.
It would also enable feature-completeness with the standard Whatsapp client, which first checks the on-phone addressbook for nicknames.

**Comment by d99kris** [comment]
Hi @hseg - thanks for suggesting a feature. I think providing external app management might be complex, but perhaps there's some simpler way to provide a "good enough" user experience. Would the ability to rename contacts (locally, within nchat) would be useful?

**Comment by hseg** [comment]
In fact, the use of an external app was meant to _reduce_ the complexity here by avoiding embedding contact-management software into nchat. The idea was to make the address book nchat uses available as eg a directory of vcard files, which could then be managed by eg abook/khard/…
Although, supporting a simple alias file like git's .mailmap would (at the potential cost of duplicating contact management efforts)  at least make day-to-day operations easier.

**Reply by d99kris** [reply]
Okay, thanks for clarifying. 👍

**Comment by hseg** [comment]
I don't know how or why, but one of my contacts who uses Whatsapp for Business suddenly has their name displayed according to my address book, not their Whatsapp nick. This was not the case this morning, and I don't think I've updated nchat in between.
I'm not sure if this means the motivating bug here has been fixed (by accident?), but thought I'd share the update.

**Reply by d99kris** [reply]
Thanks for sharing. Yeah the nchat implementation hasn't changed either recently, it still just uses the contact/chat info it receives from WhatsApp, so nchat should match what is shown in WhatsApp Web client. Maybe WhatsApp changed something on service side.
